jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/sys/DatabaseOps.java

```diff
@@ -198,6 +198,12 @@ public static void compact(DatasetGraphSwitchable container) {
             LOG.debug(String.format("Compact %s -> %s\n", db1.getFileName(), db2.getFileName()));
 
             compact(container, loc1, loc2);
+
+            if ( shouldDeleteOld ) {
+                Path loc1Path = IOX.asPath(loc1);
+                LOG.info("Deleting old database after successful compaction (old db path='" + loc1Path + "'...");
+                IOX.delete(loc1Path);
```

Comment:
`IOX.delete` (which calls JDK `Files.delete`) only delete a file or an empty directory.

See documentation for `Files.delete` - it needs a file walk to delete recursively (Don't use `FileOps.clearAll` - old code, needs updating.)

Called on a directory with files, `IOX.delete` throws `DirectoryNotEmptyException`.

If there a test for this?

Reply:
No test yet. Wanted to see how the change was going to be received first.

Reply:
I've updated this to use `Files.walk()...`.

Reply:
> No test yet. Wanted to see how the change was going to be received first.

Better to set this as "draft" if it is not ready for merging to be clear.

Reply:
test added